pallets/bridge_transfer/src/lib.rs

```diff
@@ -1,106 +1,128 @@
-
-	/// Specifies the origin check provided by the bridge for calls that can only be called by the bridge pallet
-	type BridgeOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
-
-	/// The currency mechanism.
-	type Currency: Currency<Self::AccountId>;
-
-	type BridgeTokenId: Get<ResourceId>;
-	type BridgeLotteryId: Get<ResourceId>;
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as BridgeTransfer {
-		BridgeFee get(fn bridge_fee): map hasher(opaque_blake2_256) bridge::BridgeChainId => (BalanceOf<T>, u32);
+pub use pallet::*;
+#[frame_support::pallet]
+pub mod pallet {
+	use codec::{Decode, Encode};
+	use frame_support::{
+		fail,
+		pallet_prelude::*,
+		traits::{Currency, ExistenceRequirement::AllowDeath},
+	};
+	use frame_system::pallet_prelude::*;
+	pub use pallet_bridge as bridge;
+	use sp_arithmetic::traits::SaturatedConversion;
+	use sp_core::U256;
+	use sp_std::convert::TryFrom;
+	use sp_std::prelude::*;
+
+	use pallet_mq::MessageOriginInfo;
+	use phala_pallets::pallet_mq;
+	use phala_types::messaging::{DecodedMessage, Lottery, LotteryCommand};
+
+	type ResourceId = bridge::ResourceId;
+
+	type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + bridge::Config + pallet_mq::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Specifies the origin check provided by the bridge for calls that can only be called by the bridge pallet
+		type BridgeOrigin: EnsureOrigin<Self::Origin, Success = Self::AccountId>;
+
+		/// The currency mechanism.
+		type Currency: Currency<Self::AccountId>;
+
+		#[pallet::constant]
+		type BridgeTokenId: Get<ResourceId>;
+		#[pallet::constant]
+		type BridgeLotteryId: Get<ResourceId>;
 	}
-}
 
-decl_event! {
-	pub enum Event<T>
-	where
-		Balance = BalanceOf<T>,
-	{
+	#[pallet::storage]
+	#[pallet::getter(fn bridge_fee)]
+	pub type BridgeFee<T> =
+		StorageMap<_, Blake2_256, bridge::BridgeChainId, (BalanceOf<T>, u32), ValueQuery>;
+
+	#[pallet::event]
+	#[pallet::metadata(BalanceOf<T> = "Balance")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
 		/// [chainId, min_fee, fee_scale]
-		FeeUpdated(bridge::BridgeChainId, Balance, u32),
+		FeeUpdated(bridge::BridgeChainId, BalanceOf<T>, u32),
 	}
-}
 
-decl_error! {
-	pub enum Error for Module<T: Config>{
+	#[pallet::error]
+	pub enum Error<T> {
 		InvalidTransfer,
 		InvalidCommand,
 		InvalidPayload,
 		InvalidFeeOption,
 		FeeOptionsMissiing,
 	}
-}
 
-decl_module! {
-	pub struct Module<T: Config> for enum Call where origin: T::Origin {
-		type Error = Error<T>;
-		//
-		// Initiation calls. These start a bridge transfer.
-		//
-
-		fn deposit_event() = default;
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
```

Comment:
Not needed. We don't have any hook in this pallet

Reply:
Remove it